File: service/db/src/main/java/org/delivery/db/storemenu/StoreMenuRepository.java
package org.delivery.db.storemenu;

import java.util.List;
import java.util.Optional;

import org.delivery.db.storemenu.vo.StoreMenuStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StoreMenuRepository extends JpaRepository<StoreMenu, Long> {

	Optional<StoreMenu> findFirstByIdAndStatusOrderByIdDesc(final Long id, final StoreMenuStatus status);
	List<StoreMenu> findAllByStoreIdAndStatusOrderBySequenceDesc(final Long sotreId, final StoreMenuStatus status);

}

File: chat-service/service/db/src/main/kotlin/org/delivery/db/storemenu/StoreMenuRepository.kt
package org.delivery.db.storemenu

import org.delivery.db.storemenu.vo.StoreMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreMenuRepository : JpaRepository<StoreMenu, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(id: Long?, status: StoreMenuStatus?): StoreMenu?
    fun findAllByStoreIdAndStatusOrderBySequenceDesc(storeId: Long?, status: StoreMenuStatus?): List<StoreMenu>

}
